executor/cook/executor.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python3
+
+import json
+import logging
+import os
+import re
+import subprocess
+import time
+
+from pymesos import Executor, MesosExecutorDriver, decode_data, encode_data
+from threading import Event, Lock, Thread
+
+
+def get_task_id(task):
+    """
+        :param task: The task
+        :return: The id of the task.
+    """
+    return task['task_id']['value']
+
+
+def create_status(task_id, task_state):
+    """
+        :param task_id: The task id.
+        :param task_state: The state of the task to report.
+        :return: a status dictionary that can be sent to the driver.
+    """
+    return {'task_id': {'value': task_id},
+            'state': task_state,
+            'timestamp': time.time()}
+
+
+def update_status(driver, task, task_state):
+    """
+        :param driver: The driver to send the status update to.
+        :param task: The task whose status update to send.
+        :param task_state: The state of the task which will be sent to the driver.
+        :return: Nothing.
+    """
+    task_id = get_task_id(task)
+    logging.debug('Updating task {} state to {}'.format(task_id, task_state))
+    status = create_status(task_id, task_state)
+    driver.sendStatusUpdate(status)
+
+
+def send_message(driver, message, max_message_length):
+    """
+        :param driver: The driver to send the message to.
+        :param message: The raw message to send.
+        :param max_message_length: The allowed max message length after encoding
+        :return: whether the message was successfully sent
+    """
+    logging.debug('Sending framework message {}'.format(message))
+    encoded_message = encode_data(str(message).encode('utf8'))
+    if len(encoded_message) < max_message_length:
+        driver.sendFrameworkMessage(encoded_message)
+        return True
+    else:
+        log_message_template = 'Unable to send message {} as it exceeds allowed max length of {}'
+        logging.info(log_message_template.format(message, max_message_length))
+        return False
+
+
+def launch_task(driver, task, stdout_file, stderr_file):
+    """
+        :param driver: The mesos driver to use.
+        :param task: The task to execute.
+        :param stdout_file: The file to use to redirect stdout.
+        :param stderr_file: The file to use to redirect stderr.
+        :return: The tuple containing the process launched, stdout file, and stderr file.
+    """
+    try:
+        update_status(driver, task, 'TASK_RUNNING')
+
+        command = decode_data(task['data']).decode('utf8')
+        logging.info('Command: {}'.format(command))
+        stdout = open(stdout_file, 'a+')
```

Comment:
During debugging I also used echo to generate output, these get written to stdout. Using the append mode avoided deleting those messages.

In general, the append mode should hurt and will avoid gobbling messages generated from other sources to stdout/stderr files.